extensions/guacamole-auth-quickconnect/src/main/java/org/apache/guacamole/auth/quickconnect/QuickConnectUserContext.java

```diff
@@ -0,0 +1,123 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.auth.quickconnect;
+
+import java.util.Collections;
+import org.apache.guacamole.auth.quickconnect.rest.QuickConnectREST;
+import org.apache.guacamole.GuacamoleException;
+import org.apache.guacamole.net.auth.AbstractUserContext;
+import org.apache.guacamole.net.auth.AuthenticationProvider;
+import org.apache.guacamole.net.auth.ConnectionGroup;
+import org.apache.guacamole.net.auth.Directory;
+import org.apache.guacamole.net.auth.User;
+import org.apache.guacamole.net.auth.simple.SimpleDirectory;
+import org.apache.guacamole.net.auth.simple.SimpleUser;
+
+/**
+ * A simple implementation of UserContext to support the QuickConnect
+ * extension, used for storing connections the user has created
+ * with the QuickConnect bar in the webapp.
+ */
+public class QuickConnectUserContext extends AbstractUserContext {
+
+    /**
+     * The AuthenticationProvider that created this UserContext.
+     */
+    private final AuthenticationProvider authProvider;
+
+    /**
+     * Reference to the user whose permissions dictate the configurations
+     * accessible within this UserContext.
+     */
+    private final User self;
+
+    /**
+     * The Directory with access to all connections within the root group
+     * associated with this UserContext.
+     */
+    private final QuickConnectDirectory connectionDirectory;
+
+    /**
+     * The root connection group.
+     */
+    private final ConnectionGroup rootGroup;
+
+    /**
+     * Construct a QuickConnectUserContext using the authProvider and
+     * the username.
+     *
+     * @param authProvider
+     *     The authentication provider module instantiating this
+     *     this class.
+     *
+     * @param username
+     *     The name of the user logging in and using this class.
+     */
+    public QuickConnectUserContext(AuthenticationProvider authProvider,
+            String username) {
+
+        // Initialize the rootGroup to a basic connection group with a
+        // single root identifier.
+        this.rootGroup = new QuickConnectionGroup(
+            DEFAULT_ROOT_CONNECTION_GROUP,
+            DEFAULT_ROOT_CONNECTION_GROUP
+        );
+
+        // Initialize the user to a SimpleUser with the provided username,
+        // no connections, and the single root group.
+        this.self = new SimpleUser(username,
+            Collections.<String>emptyList(),
```

Comment:
This will only be correct when no connections exist. Once at least one connection is added, the permissions stored within this `SimpleUser` will not match up with the connections they are supposed to be able to read.

I'd recommend instead returning a new `SimpleUser` from `self()`, using `connectionDirectory.getIdentifiers()` to pull the set of readable connections.

Reply:
I believe I've done this as you suggested.